persist/fs/commitlog/commit_log_test.go

```diff
@@ -330,6 +346,100 @@ func TestCommitLogWrite(t *testing.T) {
 	assertCommitLogWritesByIterating(t, commitLog, writes)
 }
 
+func TestReadCommitLogMissingMetadata(t *testing.T) {
+	readConc := 4
+	// Make sure we're not leaking goroutines
+	defer leaktest.CheckTimeout(t, 10*time.Second)()
+
+	opts, scope := newTestOptions(t, overrides{
+		strategy: StrategyWriteWait,
+	})
+	// Set read concurrency so that the parallel path is definitely tested
+	opts.SetReadConcurrency(readConc)
+	defer cleanup(t, opts)
+
+	// Replace bitset in writer with one that configurably returns true or false
+	// depending on the series
+	commitLog := newTestCommitLog(t, opts)
+	writer := commitLog.writer.(*writer)
+	mockBitSet := &mockBitSet{
+		indexTestReturn: map[uint]bool{},
+	}
+
+	// Generate fake series, where approximately half will be missing metadata.
+	// This works because the commitlog writer uses the bitset to determine if
+	// the metadata for a particular series had already been written to disk.
+	allSeries := []Series{}
+	for i := 0; i < 200; i++ {
+		willNotHaveMetadata := !(i%2 == 0)
+		allSeries = append(allSeries, testSeries(uint64(i), "hax", uint32(i%100)))
+		mockBitSet.indexTestReturn[uint(i)] = willNotHaveMetadata
+	}
+	writer.seen = mockBitSet
+
+	// Generate fake writes for each of the series
+	writes := []testWrite{}
+	for _, series := range allSeries {
+		for i := 0; i < 10; i++ {
+			writes = append(writes, testWrite{series, time.Now(), rand.Float64(), xtime.Second, []byte{1, 2, 3}, nil})
+		}
+	}
+
+	// Call write sync
+	writeCommitLogs(t, scope, commitLog, writes).Wait()
+
+	// Close the commit log and consequently flush
+	assert.NoError(t, commitLog.Close())
+
+	// Make sure we don't panic / deadlock
+	iter, err := commitLog.Iter()
+	assert.NoError(t, err)
+	for iter.Next() {
+	}
+	iter.Close()
```

Comment:
do you need to check the return code for `iter.Err()`, and the other close methods too?

Reply:
added it